### PR TITLE
doc/start: remove journal info from hardware recs

### PR DESCRIPTION
**Pull request by zdover23**
This PR removes mentions of journaling from the hardware
recommendations.

Journaling was a FileStore-related practice. BlueStore is
the default backend for Ceph OSDs and has been since
Luminous. The documentation should reflect that.

Signed-off-by: Zac Dover <zac.dover@gmail.com>
(cherry picked from commit 276bbd8f2b516993813fedfdfd1371e87bb0d44a)

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: doc/start/hardware-recommendations.rst [hunk 1]
```diff
@@ -178,36 +163,15 @@ hard disk drive.
...
-  or better than a hard disk drive when writing data. Inexpensive SSDs may 
-  introduce write latency even as they accelerate access time, because 
-  sometimes high performance hard drives can write as fast or faster than 
-  some of the more economical SSDs available on the market!
-  
-- **Sequential Writes:** When you store multiple journals on an SSD you must 
-  consider the sequential write limitations of the SSD too, since they may be 
-  handling requests to write to multiple OSD journals simultaneously.
-
-- **Partition Alignment:** A common problem with SSD performance is that
```
Comment (anthonyeleven) [inline doc/start/hardware-recommendations.rst:207]: Alignment is still a concern with BlueStore, I might suggest leaving this section in.

File: doc/start/hardware-recommendations.rst [hunk 2]
```diff
@@ -151,20 +146,10 @@ overall system performance--especially during recovery. We recommend using a
...
 You may run multiple Ceph OSD Daemons per SAS / SATA drive, but this will likely
-lead to resource contention and diminish the overall throughput. You may store a
-journal and object data on the same drive, but this may increase the time it
-takes to journal a write and ACK to the client. Ceph must write to the journal
-before it can ACK the write.
-
-Ceph best practices dictate that you should run operating systems, OSD data and
-OSD journals on separate drives.
-
+lead to resource contention and diminish the overall throughput.
```
Comment (anthonyeleven) [inline doc/start/hardware-recommendations.rst:153]: I suggest leaving something in about separating the OS and OSD data.  Mon/mgr data IMHO is probably fine on an OS drive.

**Review by anthonyeleven — APPROVED** [review]
I left a couple of non-binding comments, process or discard at will.

File: doc/start/hardware-recommendations.rst (hunk 1 above)
Comment (zdover23) [reply doc/start/hardware-recommendations.rst:207]: @jdurgin -- I need a judgment call on this one. If this is true, then we'll have to restore this to the master branch.

Ditto for Anthony's other comments below.

**Comment by jdurgin** [comment]
Both @anthonyeleven's comments are correct - we should fix those in a separate PR in master

**Comment by zdover23** [comment]
> Both @anthonyeleven's comments are correct - we should fix those in a separate PR in master

Roger that. I'll proceed with this merge. We'll make Anthony's suggested changes in a separate PR (and backport them using the same procedure I used here).

**Comment by zdover23** [comment]
https://github.com/ceph/ceph/pull/45139

The refinements discussed https://github.com/ceph/ceph/pull/45123#issuecomment-1049311216 are made in the pull request listed above.

**Pull request merged**